localstack-core/localstack/services/lambda_/provider.py

```diff
@@ -1990,6 +2071,63 @@ def update_event_source_mapping(
         state.event_source_mappings[uuid] = event_source_mapping
         return {**event_source_mapping, **temp_params}
 
+    def update_event_source_mapping_v2(
+        self,
+        context: RequestContext,
+        request: UpdateEventSourceMappingRequest,
+    ) -> EventSourceMappingConfiguration:
+        # TODO: test and implement this properly (quite complex with many validations and limitations!)
+        LOG.warning(
+            "Updating Lambda Event Source Mapping is in experimental state and not yet fully tested."
+        )
+        state = lambda_stores[context.account_id][context.region]
+        request_data = {**request}
+        uuid = request_data.pop("UUID", None)
+        if not uuid:
+            raise ResourceNotFoundException(
+                "The resource you requested does not exist.", Type="User"
+            )
+        old_event_source_mapping = state.event_source_mappings.get(uuid)
+        if old_event_source_mapping is None:
+            raise ResourceNotFoundException(
+                "The resource you requested does not exist.", Type="User"
+            )  # TODO: test?
+
+        # remove the FunctionName field
+        function_name_or_arn = request_data.pop("FunctionName", None)
+
+        # normalize values to overwrite
+        event_source_mapping = {
+            **old_event_source_mapping,
+            **request_data,
+        }
```

Comment:
Could use `|` here, personally I think it is more readable.

Reply:
Fully agree 👍
(that was existing code, so improved it in both places)